File: Backend/src/main/java/com/sencarmarket/module/garage/controller/GarageController.java
package com.sencarmarket.module.garage.controller;

import com.sencarmarket.module.commun.dto.PaginatedResponse;
import com.sencarmarket.module.garage.dto.*;
import com.sencarmarket.module.garage.service.GarageService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.UUID;

/**
 * Contrôleur REST pour la gestion des garages
 */
@RestController
@RequestMapping("/api/garages")
@RequiredArgsConstructor
@Slf4j
public class GarageController {

    private final GarageService garageService;

    // ========== GARAGE ==========

    /**
     * Créer un nouveau garage
     * POST /api/garages
     */
    @PostMapping
    public ResponseEntity<GarageResponse> createGarage(
            @Valid @RequestBody CreateGarageRequest request,
            @RequestHeader("X-User-Id") UUID proprietaireId) {
        log.info("POST /api/garages - Creating garage '{}' for user {}", request.getNom(), proprietaireId);
        GarageResponse response = garageService.createGarage(request, proprietaireId);
        return ResponseEntity.status(HttpStatus.CREATED).body(response);
    }

    /**
     * Obtenir un garage par ID
     * GET /api/garages/{id}
     */
    @GetMapping("/{id}")
    public ResponseEntity<GarageResponse> getGarageById(@PathVariable UUID id) {
        log.debug("GET /api/garages/{}", id);
        return ResponseEntity.ok(garageService.getGarageById(id));
    }

    /**
     * Obtenir tous les garages (pagination)
     * GET /api/garages
     */
    @GetMapping
    public ResponseEntity<PaginatedResponse<GarageResponse>> getAllGarages(
            @RequestParam(defaultValue = "0") int page,
            @RequestParam(defaultValue = "10") int size) {
        log.debug("GET /api/garages - page: {}, size: {}", page, size);
        return ResponseEntity.ok(garageService.getAllGarages(page, size));
    }

    /**
     * Obtenir les garages actifs
     * GET /api/garages/actifs
     */
    @GetMapping("/actifs")
    public ResponseEntity<PaginatedResponse<GarageResponse>> getActiveGarages(
            @RequestParam(defaultValue = "0") int page,
            @RequestParam(defaultValue = "10") int size) {
        log.debug("GET /api/garages/actifs - page: {}, size: {}", page, size);
        return ResponseEntity.ok(garageService.getActiveGarages(page, size));
    }

    /**
     * Obtenir les garages en attente de validation (admin)
     * GET /api/garages/en-attente
     */
    @GetMapping("/en-attente")
    public ResponseEntity<PaginatedResponse<GarageResponse>> getGaragesEnAttente(
            @RequestParam(defaultValue = "0") int page,
            @RequestParam(defaultValue = "10") int size) {
        log.debug("GET /api/garages/en-attente - page: {}, size: {}", page, size);
        return ResponseEntity.ok(garageService.getGaragesEnAttente(page, size));
    }

    /**
     * Obtenir les garages d'un propriétaire
     * GET /api/garages/proprietaire/{proprietaireId}
     */
    @GetMapping("/proprietaire/{proprietaireId}")
    public ResponseEntity<List<GarageResponse>> getGaragesByProprietaire(@PathVariable UUID proprietaireId) {
        log.debug("GET /api/garages/proprietaire/{}", proprietaireId);
        return ResponseEntity.ok(garageService.getGaragesByProprietaire(proprietaireId));
    }

    /**
     * Rechercher des garages par ville
     * GET /api/garages/search/ville
     */
    @GetMapping("/search/ville")
    public ResponseEntity<List<GarageResponse>> searchByVille(@RequestParam String ville) {
        log.debug("GET /api/garages/search/ville?ville={}", ville);
        return ResponseEntity.ok(garageService.searchByLocalisation(ville));
    }

    /**
     * Rechercher des garages par proximité
     * GET /api/garages/search/proximity
     */
    @GetMapping("/search/proximity")
    public ResponseEntity<List<GarageResponse>> searchByProximity(
            @RequestParam Double latitude,
            @RequestParam Double longitude,
            @RequestParam(defaultValue = "10") Double rayonKm) {
        log.debug("GET /api/garages/search/proximity?lat={}&lon={}&rayon={}", latitude, longitude, rayonKm);
        return ResponseEntity.ok(garageService.searchByProximity(latitude, longitude, rayonKm));
    }

    /**
     * Rechercher des garages
     * GET /api/garages/search
     */
    @GetMapping("/search")
    public ResponseEntity<List<GarageResponse>> searchGarages(@RequestParam String q) {
        log.debug("GET /api/garages/search?q={}", q);
        return ResponseEntity.ok(garageService.searchGarages(q));
    }

    /**
     * Mettre à jour un garage
     * PUT /api/garages/{id}
     */
    @PutMapping("/{id}")
    public ResponseEntity<GarageResponse> updateGarage(
            @PathVariable UUID id,
            @Valid @RequestBody CreateGarageRequest request) {
        log.info("PUT /api/garages/{}", id);
        return ResponseEntity.ok(garageService.updateGarage(id, request));
    }

    /**
     * Supprimer un garage
     * DELETE /api/garages/{id}
     */
    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deleteGarage(@PathVariable UUID id) {
        log.info("DELETE /api/garages/{}", id);
        garageService.deleteGarage(id);
        return ResponseEntity.noContent().build();
    }

    /**
     * Valider un garage (admin)
     * POST /api/garages/{id}/validate
     */
    @PostMapping("/{id}/validate")
    public ResponseEntity<GarageResponse> validerGarage(
            @PathVariable UUID id,
            @Valid @RequestBody ValidationGarageRequest request) {
        log.info("POST /api/garages/{}/validate", id);
        return ResponseEntity.ok(garageService.validerGarage(id, request));
    }

    /**
     * Mettre à jour le logo
     * PUT /api/garages/{id}/logo
     */
    @PutMapping("/{id}/logo")
    public ResponseEntity<GarageResponse> updateLogo(
            @PathVariable UUID id,
            @RequestBody String logoUrl) {
        log.info("PUT /api/garages/{}/logo", id);
        return ResponseEntity.ok(garageService.updateLogo(id, logoUrl));
    }

    // ========== SERVICES GARAGE ==========

    /**
     * Créer un service
     * POST /api/garages/services
     */
    @PostMapping("/services")
    public ResponseEntity<ServiceGarageResponse> createService(@Valid @RequestBody CreateServiceGarageRequest request) {
        log.info("POST /api/garages/services - Creating service '{}'", request.getNom());
        return ResponseEntity.status(HttpStatus.CREATED).body(garageService.createService(request));
    }

    /**
     * Obtenir tous les services
     * GET /api/garages/services
     */
    @GetMapping("/services")
    public ResponseEntity<List<ServiceGarageResponse>> getAllServices() {
        log.debug("GET /api/garages/services");
        return ResponseEntity.ok(garageService.getAllServices());
    }

    /**
     * Obtenir un service par ID
     * GET /api/garages/services/{id}
     */
    @GetMapping("/services/{id}")
    public ResponseEntity<ServiceGarageResponse> getServiceById(@PathVariable UUID id) {
        log.debug("GET /api/garages/services/{}", id);
        return ResponseEntity.ok(garageService.getServiceById(id));
    }

    /**
     * Mettre à jour un service
     * PUT /api/garages/services/{id}
     */
    @PutMapping("/services/{id}")
    public ResponseEntity<ServiceGarageResponse> updateService(
            @PathVariable UUID id,
            @Valid @RequestBody CreateServiceGarageRequest request) {
        log.info("PUT /api/garages/services/{}", id);
        return ResponseEntity.ok(garageService.updateService(id, request));
    }

    /**
     * Supprimer un service
     * DELETE /api/garages/services/{id}
     */
    @DeleteMapping("/services/{id}")
    public ResponseEntity<Void> deleteService(@PathVariable UUID id) {
        log.info("DELETE /api/garages/services/{}", id);
        garageService.deleteService(id);
        return ResponseEntity.noContent().build();
    }

    // ========== ASSOCIATIONS GARAGE-SERVICE ==========

    /**
     * Associer un service à un garage
     * POST /api/garages/{garageId}/services
     */
    @PostMapping("/{garageId}/services")
    public ResponseEntity<GarageServiceResponse> associateService(
            @PathVariable UUID garageId,
            @Valid @RequestBody AssociateServiceRequest request) {
        log.info("POST /api/garages/{}/services - Associating service", garageId);
        request.setGarageId(garageId);
        return ResponseEntity.status(HttpStatus.CREATED).body(garageService.associateService(request));
    }

    /**
     * Obtenir les services d'un garage
     * GET /api/garages/{garageId}/services
     */
    @GetMapping("/{garageId}/services")
    public ResponseEntity<List<GarageServiceResponse>> getServicesByGarage(@PathVariable UUID garageId) {
        log.debug("GET /api/garages/{}/services", garageId);
        return ResponseEntity.ok(garageService.getServicesByGarage(garageId));
    }

    /**
     * Supprimer l'association service-garage
     * DELETE /api/garages/{garageId}/services/{serviceId}
     */
    @DeleteMapping("/{garageId}/services/{serviceId}")
    public ResponseEntity<Void> disassociateService(
            @PathVariable UUID garageId,
            @PathVariable UUID serviceId) {
        log.info("DELETE /api/garages/{}/services/{}", garageId, serviceId);
        garageService.disassociateService(garageId, serviceId);
        return ResponseEntity.noContent().build();
    }
}

File: Backend/src/modules/garage/garage.controller.ts
import {
  Body,
  Controller,
  Delete,
  Get,
  HttpCode,
  HttpStatus,
  Param,
  ParseFloatPipe,
  ParseIntPipe,
  ParseUUIDPipe,
  Post,
  Put,
  Query,
  UseGuards,
} from '@nestjs/common';
import { ApiBearerAuth, ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';

import { CurrentUser } from '../../common/decorators/current-user.decorator';
import { PaginatedResponseDto } from '../../common/dto/paginated-response.dto';
import { JwtAuthGuard } from '../../common/guards/jwt-auth.guard';
import type { AuthenticatedUser } from '../../common/types/authenticated-user.type';
import { ApiErrorResponseDto } from '../auth/dto/api-error-response.dto';

import { AssociateServiceRequestDto } from './dto/associate-service-request.dto';
import { CreateGarageRequestDto } from './dto/create-garage-request.dto';
import { CreateServiceGarageRequestDto } from './dto/create-service-garage-request.dto';
import { GarageResponseDto } from './dto/garage-response.dto';
import { GarageServiceResponseDto } from './dto/garage-service-response.dto';
import { ServiceGarageResponseDto } from './dto/service-garage-response.dto';
import { ValidationGarageRequestDto } from './dto/validation-garage-request.dto';
import { GarageService } from './garage.service';

@ApiTags('Garages')
@ApiBearerAuth()
@Controller('garages')
export class GarageController {
  constructor(private readonly service: GarageService) {}

  @Post()
  @UseGuards(JwtAuthGuard)
  @HttpCode(HttpStatus.CREATED)
  @ApiOperation({ summary: 'Créer un garage' })
  @ApiResponse({ status: 201, type: GarageResponseDto, description: 'Garage créé avec succès' })
  @ApiResponse({ status: 400, type: ApiErrorResponseDto, description: 'Données invalides' })
  @ApiResponse({ status: 401, type: ApiErrorResponseDto, description: 'Non autorisé - Token invalide ou expiré' })
  @ApiResponse({ status: 403, type: ApiErrorResponseDto, description: 'Interdit - Accès refusé' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  createGarage(
    @Body() request: CreateGarageRequestDto,
    @CurrentUser() user: AuthenticatedUser,
  ): Promise<GarageResponseDto> {
    return this.service.createGarage(request, user);
  }

  @Get()
  @ApiOperation({ summary: 'Liste des garages' })
  @ApiResponse({ status: 200, type: PaginatedResponseDto, description: 'Garages récupérés avec succès' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  getAllGarages(
    @Query('page', new ParseIntPipe({ optional: true })) page = 0,
    @Query('size', new ParseIntPipe({ optional: true })) size = 10,
  ): Promise<PaginatedResponseDto<GarageResponseDto>> {
    return this.service.getAllGarages(page, size);
  }

  @Get('actifs')
  @ApiOperation({ summary: 'Garages actifs' })
  @ApiResponse({ status: 200, type: PaginatedResponseDto, description: 'Garages actifs récupérés avec succès' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  getActiveGarages(
    @Query('page', new ParseIntPipe({ optional: true })) page = 0,
    @Query('size', new ParseIntPipe({ optional: true })) size = 10,
  ): Promise<PaginatedResponseDto<GarageResponseDto>> {
    return this.service.getActiveGarages(page, size);
  }

  @Get('en-attente')
  @UseGuards(JwtAuthGuard)
  @ApiOperation({ summary: 'Garages en attente' })
  @ApiResponse({ status: 200, type: PaginatedResponseDto, description: 'Garages en attente récupérés avec succès' })
  @ApiResponse({ status: 401, type: ApiErrorResponseDto, description: 'Non autorisé - Token invalide ou expiré' })
  @ApiResponse({ status: 403, type: ApiErrorResponseDto, description: 'Interdit - Accès réservé aux administrateurs' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  getGaragesEnAttente(
    @CurrentUser() user: AuthenticatedUser,
    @Query('page', new ParseIntPipe({ optional: true })) page = 0,
    @Query('size', new ParseIntPipe({ optional: true })) size = 10,
  ): Promise<PaginatedResponseDto<GarageResponseDto>> {
    return this.service.getGaragesEnAttente(page, size, user);
  }

  @Get('proprietaire/:proprietaireId')
  @UseGuards(JwtAuthGuard)
  @ApiOperation({ summary: 'Garages d\'un propriétaire' })
  @ApiResponse({ status: 200, type: GarageResponseDto, isArray: true, description: 'Garages du propriétaire récupérés avec succès' })
  @ApiResponse({ status: 401, type: ApiErrorResponseDto, description: 'Non autorisé - Token invalide ou expiré' })
  @ApiResponse({ status: 404, type: ApiErrorResponseDto, description: 'Propriétaire non trouvé' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  getGaragesByProprietaire(
    @Param('proprietaireId', new ParseUUIDPipe()) proprietaireId: string,
    @CurrentUser() user: AuthenticatedUser,
  ): Promise<GarageResponseDto[]> {
    return this.service.getGaragesByProprietaire(proprietaireId, user);
  }

  @Get('search/ville')
  @ApiOperation({ summary: 'Rechercher par ville' })
  @ApiResponse({ status: 200, type: GarageResponseDto, isArray: true, description: 'Garages trouvés par ville' })
  @ApiResponse({ status: 400, type: ApiErrorResponseDto, description: 'Paramètre ville requis' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  searchByVille(@Query('ville') ville: string): Promise<GarageResponseDto[]> {
    return this.service.searchByLocalisation(ville);
  }

  @Get('search/proximity')
  @ApiOperation({ summary: 'Rechercher par proximité' })
  @ApiResponse({ status: 200, type: GarageResponseDto, isArray: true, description: 'Garages trouvés par proximité' })
  @ApiResponse({ status: 400, type: ApiErrorResponseDto, description: 'Paramètres latitude et longitude requis' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  searchByProximity(
    @Query('latitude', ParseFloatPipe) latitude: number,
    @Query('longitude', ParseFloatPipe) longitude: number,
    @Query('rayonKm', new ParseFloatPipe({ optional: true })) rayonKm = 10,
  ): Promise<GarageResponseDto[]> {
    return this.service.searchByProximity(latitude, longitude, rayonKm);
  }

  @Get('search')
  @ApiOperation({ summary: 'Rechercher des garages' })
  @ApiResponse({ status: 200, type: GarageResponseDto, isArray: true, description: 'Garages trouvés' })
  @ApiResponse({ status: 400, type: ApiErrorResponseDto, description: 'Paramètre de recherche requis' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  searchGarages(@Query('q') q: string): Promise<GarageResponseDto[]> {
    return this.service.searchGarages(q);
  }

  @Post('services')
  @UseGuards(JwtAuthGuard)
  @HttpCode(HttpStatus.CREATED)
  @ApiOperation({ summary: 'Créer un service garage' })
  @ApiResponse({ status: 201, type: ServiceGarageResponseDto, description: 'Service garage créé avec succès' })
  @ApiResponse({ status: 400, type: ApiErrorResponseDto, description: 'Données invalides' })
  @ApiResponse({ status: 401, type: ApiErrorResponseDto, description: 'Non autorisé - Token invalide ou expiré' })
  @ApiResponse({ status: 403, type: ApiErrorResponseDto, description: 'Interdit - Accès refusé' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  createService(
    @Body() request: CreateServiceGarageRequestDto,
    @CurrentUser() user: AuthenticatedUser,
  ): Promise<ServiceGarageResponseDto> {
    return this.service.createService(request, user);
  }

  @Get('services')
  @ApiOperation({ summary: 'Liste des services' })
  @ApiResponse({ status: 200, type: ServiceGarageResponseDto, isArray: true, description: 'Services récupérés avec succès' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  getAllServices(): Promise<ServiceGarageResponseDto[]> {
    return this.service.getAllServices();
  }

  @Get('services/:id')
  @ApiOperation({ summary: 'Obtenir un service par ID' })
  @ApiResponse({ status: 200, type: ServiceGarageResponseDto, description: 'Service trouvé' })
  @ApiResponse({ status: 404, type: ApiErrorResponseDto, description: 'Service non trouvé' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  getServiceById(@Param('id', new ParseUUIDPipe()) id: string): Promise<ServiceGarageResponseDto> {
    return this.service.getServiceById(id);
  }

  @Post(':garageId/services')
  @UseGuards(JwtAuthGuard)
  @HttpCode(HttpStatus.CREATED)
  @ApiOperation({ summary: 'Associer un service à un garage' })
  @ApiResponse({ status: 201, type: GarageServiceResponseDto, description: 'Service associé au garage avec succès' })
  @ApiResponse({ status: 400, type: ApiErrorResponseDto, description: 'Données invalides' })
  @ApiResponse({ status: 401, type: ApiErrorResponseDto, description: 'Non autorisé - Token invalide ou expiré' })
  @ApiResponse({ status: 403, type: ApiErrorResponseDto, description: 'Interdit - Accès refusé' })
  @ApiResponse({ status: 404, type: ApiErrorResponseDto, description: 'Garage ou service non trouvé' })
  @ApiResponse({ status: 409, type: ApiErrorResponseDto, description: 'Conflict - Service déjà associé au garage' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  associateService(
    @Param('garageId', new ParseUUIDPipe()) garageId: string,
    @Body() request: AssociateServiceRequestDto,
    @CurrentUser() user: AuthenticatedUser,
  ): Promise<GarageServiceResponseDto> {
    return this.service.associateService(garageId, request, user);
  }

  @Get(':garageId/services')
  @ApiOperation({ summary: 'Services d\'un garage' })
  @ApiResponse({ status: 200, type: GarageServiceResponseDto, isArray: true, description: 'Services du garage récupérés avec succès' })
  @ApiResponse({ status: 404, type: ApiErrorResponseDto, description: 'Garage non trouvé' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  getServicesByGarage(@Param('garageId', new ParseUUIDPipe()) garageId: string): Promise<GarageServiceResponseDto[]> {
    return this.service.getServicesByGarage(garageId);
  }

  @Delete(':garageId/services/:serviceId')
  @UseGuards(JwtAuthGuard)
  @HttpCode(HttpStatus.NO_CONTENT)
  @ApiOperation({ summary: 'Supprimer association service-garage' })
  @ApiResponse({ status: 204, description: 'Association supprimée avec succès' })
  @ApiResponse({ status: 401, type: ApiErrorResponseDto, description: 'Non autorisé - Token invalide ou expiré' })
  @ApiResponse({ status: 403, type: ApiErrorResponseDto, description: 'Interdit - Accès refusé' })
  @ApiResponse({ status: 404, type: ApiErrorResponseDto, description: 'Association non trouvée' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  async disassociateService(
    @Param('garageId', new ParseUUIDPipe()) garageId: string,
    @Param('serviceId', new ParseUUIDPipe()) serviceId: string,
    @CurrentUser() user: AuthenticatedUser,
  ): Promise<void> {
    await this.service.disassociateService(garageId, serviceId, user);
  }

  @Get(':id')
  @ApiOperation({ summary: 'Obtenir un garage par ID' })
  @ApiResponse({ status: 200, type: GarageResponseDto, description: 'Garage trouvé' })
  @ApiResponse({ status: 404, type: ApiErrorResponseDto, description: 'Garage non trouvé' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  getGarageById(@Param('id', new ParseUUIDPipe()) id: string): Promise<GarageResponseDto> {
    return this.service.getGarageById(id);
  }

  @Put(':id')
  @UseGuards(JwtAuthGuard)
  @ApiOperation({ summary: 'Mettre à jour un garage' })
  @ApiResponse({ status: 200, type: GarageResponseDto, description: 'Garage mis à jour avec succès' })
  @ApiResponse({ status: 400, type: ApiErrorResponseDto, description: 'Données invalides' })
  @ApiResponse({ status: 401, type: ApiErrorResponseDto, description: 'Non autorisé - Token invalide ou expiré' })
  @ApiResponse({ status: 403, type: ApiErrorResponseDto, description: 'Interdit - Vous n\'êtes pas propriétaire du garage' })
  @ApiResponse({ status: 404, type: ApiErrorResponseDto, description: 'Garage non trouvé' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  updateGarage(
    @Param('id', new ParseUUIDPipe()) id: string,
    @Body() request: CreateGarageRequestDto,
    @CurrentUser() user: AuthenticatedUser,
  ): Promise<GarageResponseDto> {
    return this.service.updateGarage(id, request, user);
  }

  @Delete(':id')
  @UseGuards(JwtAuthGuard)
  @HttpCode(HttpStatus.NO_CONTENT)
  @ApiOperation({ summary: 'Supprimer un garage' })
  @ApiResponse({ status: 204, description: 'Garage supprimé avec succès' })
  @ApiResponse({ status: 401, type: ApiErrorResponseDto, description: 'Non autorisé - Token invalide ou expiré' })
  @ApiResponse({ status: 403, type: ApiErrorResponseDto, description: 'Interdit - Vous n\'êtes pas propriétaire du garage' })
  @ApiResponse({ status: 404, type: ApiErrorResponseDto, description: 'Garage non trouvé' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  async deleteGarage(
    @Param('id', new ParseUUIDPipe()) id: string,
    @CurrentUser() user: AuthenticatedUser,
  ): Promise<void> {
    await this.service.deleteGarage(id, user);
  }

  @Post(':id/validate')
  @UseGuards(JwtAuthGuard)
  @ApiOperation({ summary: 'Valider un garage' })
  @ApiResponse({ status: 200, type: GarageResponseDto, description: 'Garage validé avec succès' })
  @ApiResponse({ status: 400, type: ApiErrorResponseDto, description: 'Données de validation invalides' })
  @ApiResponse({ status: 401, type: ApiErrorResponseDto, description: 'Non autorisé - Token invalide ou expiré' })
  @ApiResponse({ status: 403, type: ApiErrorResponseDto, description: 'Interdit - Accès réservé aux administrateurs' })
  @ApiResponse({ status: 404, type: ApiErrorResponseDto, description: 'Garage non trouvé' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  validerGarage(
    @Param('id', new ParseUUIDPipe()) id: string,
    @Body() request: ValidationGarageRequestDto,
    @CurrentUser() user: AuthenticatedUser,
  ): Promise<GarageResponseDto> {
    return this.service.validerGarage(id, request, user);
  }

  @Put(':id/logo')
  @UseGuards(JwtAuthGuard)
  @ApiOperation({ summary: 'Mettre à jour le logo' })
  @ApiResponse({ status: 200, type: GarageResponseDto, description: 'Logo mis à jour avec succès' })
  @ApiResponse({ status: 400, type: ApiErrorResponseDto, description: 'URL du logo invalide' })
  @ApiResponse({ status: 401, type: ApiErrorResponseDto, description: 'Non autorisé - Token invalide ou expiré' })
  @ApiResponse({ status: 403, type: ApiErrorResponseDto, description: 'Interdit - Vous n\'êtes pas propriétaire du garage' })
  @ApiResponse({ status: 404, type: ApiErrorResponseDto, description: 'Garage non trouvé' })
  @ApiResponse({ status: 500, type: ApiErrorResponseDto, description: 'Erreur serveur interne' })
  updateLogo(
    @Param('id', new ParseUUIDPipe()) id: string,
    @Body('logoUrl') logoUrl: string,
    @CurrentUser() user: AuthenticatedUser,
  ): Promise<GarageResponseDto> {
    return this.service.updateLogo(id, logoUrl, user);
  }
}
